src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -663,6 +663,14 @@ private void TakeIncrementalSnapshot(BoundNode node)
             _snapshotBuilderOpt?.TakeIncrementalSnapshot(node, State);
         }
 
+        private void SetUpdatedSymbol(BoundNode node, Symbol originalSymbol, Symbol updatedSymbol)
+        {
+            if (_updatedSymbolMapOpt is object && (object)originalSymbol != updatedSymbol)
```

Comment:
>(object)originalSymbol != updatedSymbol [](start = 50, length = 39)

Should this compare against the current value in the map?

Reply:
I don't see why. We're not doing that for the `Type` field, not sure why we'd do that here

---
In reply to: [319280480](https://github.com/dotnet/roslyn/pull/38354#discussion_r319280480) [](ancestors = 319280480)

Reply:
Is it possible the same (node, originalSymbol) is updated twice, the first time to a distinct value, and the second time to the original value? If so, the map will contain the value from the first update.

---
In reply to: [319297666](https://github.com/dotnet/roslyn/pull/38354#discussion_r319297666) [](ancestors = 319297666,319280480)

Reply:
Technically, I guess? But again, we'd have the same concerns for the `Type` map. The original symbol should be the symbol that comes from the node itself, so if we're setting it back to the original value original and updated would be reference equal and we wouldn't set it anyway.

---
In reply to: [319299146](https://github.com/dotnet/roslyn/pull/38354#discussion_r319299146) [](ancestors = 319299146,319297666,319280480)

Reply:
I haven't come up with an example.

---
In reply to: [319300301](https://github.com/dotnet/roslyn/pull/38354#discussion_r319300301) [](ancestors = 319300301,319299146,319297666,319280480)